src/lib/viewers/ResultsViewer.ts

```diff
@@ -65,18 +65,42 @@ export class ResultsDetailDisplayer extends AbstractResultsDisplayer {
 			'summary.detail.violation-header',
 			[idx + 1, rule.getName()]
 		);
-		const body = {
-			severity: `${sev.valueOf()} (${SeverityLevel[sev]})`,
-			engine: rule.getEngineName(),
-			message: violation.getMessage(),
-			location: `${primaryLocation.getFile()}:${primaryLocation.getStartLine()}:${primaryLocation.getStartColumn()}`,
-			resources: violation.getResourceUrls().join(',')
-		};
-		const keys = ['severity', 'engine', 'message', 'location', 'resources'];
-		return toStyledHeaderAndBody(header, body, keys);
+		if (violation.getCodeLocations().length > 1) {
+			const body = {
+				severity: `${sev.valueOf()} (${SeverityLevel[sev]})`,
+				engine: rule.getEngineName(),
+				message: violation.getMessage(),
+				locations: stringifyLocations(violation.getCodeLocations(), violation.getPrimaryLocationIndex()),
+				resources: violation.getResourceUrls().join(',')
+			};
+			const keys = ['severity', 'engine', 'message', 'locations', 'resources'];
+			return toStyledHeaderAndBody(header, body, keys);
+		} else {
+			const body = {
+				severity: `${sev.valueOf()} (${SeverityLevel[sev]})`,
+				engine: rule.getEngineName(),
+				message: violation.getMessage(),
+				location: `${primaryLocation.getFile()}:${primaryLocation.getStartLine()}:${primaryLocation.getStartColumn()}`,
+				resources: violation.getResourceUrls().join(',')
+			};
+			const keys = ['severity', 'engine', 'message', 'location', 'resources'];
+			return toStyledHeaderAndBody(header, body, keys);
+		}
 	}
 }
 
+function stringifyLocations(codeLocations: CodeLocation[], primaryIndex: number): string[] {
+	const locationStrings: string[] = [];
+
+	codeLocations.forEach((loc, idx) => {
+		const commentPortion: string = loc.getComment() ? ` ${loc.getComment()}` : '';
+		const rawLocationString: string = `${loc.getFile()}:${loc.getStartLine()}:${loc.getStartColumn()}${commentPortion}`;
+		locationStrings.push(idx === primaryIndex ? `**${rawLocationString}**` : rawLocationString);
+	});
+
+	return locationStrings;
+}
```

Comment:
There is an false assumption here that if the locations length is 1 then there is no comment. This may not be the case with flowtest or with other engines in the future.

So can we instead have on line 83
```
    location: stringifyLocation(primaryLocation)
```
    
and then here we would have something like:
```
function stringifyLocations(codeLocations: CodeLocation[], primaryIndex:number): string[] {
    codeLocations.map(stringifyLocations);
    codeLocations[primaryIndex] = `**${codeLocations[primaryIndex]}**`;
}

function stringifyLocation(loc: CodeLocation): string {
    const commentPortion: string = loc.getComment() ? ` ${loc.getComment()}` : '';
	return `${loc.getFile()}:${loc.getStartLine()}:${loc.getStartColumn()}${commentPortion}`;
}
```

Reply:
Counterpoint: If there's only one location, then any comment that location might have is likely to be redundant with the top-level violation message. I'm trying to imagine a reason that they'd contain completely different information and nothing's coming to mind.

Reply:
I see your point... so the engine authors really shouldn't be doing this. But in the event they do - and they may have a legitimate reason for it... do we really want to prevent their comment from being seen? So best to code the clients to the core api and what it can can't do instead of assuming the current use cases will be the same as the future use cases.